ipam/actor.go

```diff
@@ -0,0 +1,165 @@
+package ipam
+
+import (
+	"fmt"
+	"github.com/zettio/weave/router"
+	"net"
+	"time"
+)
+
+// Start runs the allocator goroutine
+func (alloc *Allocator) Start() {
+	actionChan := make(chan func(), router.ChannelSize)
+	alloc.actionChan = actionChan
+	go alloc.actorLoop(actionChan)
+}
+
+// Actor client API
+
+// GetFor (Sync) - get IP address for container with given name
+func (alloc *Allocator) GetFor(ident string, cancelChan <-chan bool) net.IP {
```

Comment:
ah ok. No, in that case it's not worth the hassle. Isn't it great that the go standard libraries don't follow their own best practices?